### PR TITLE
[Xcode 11.4] Remove -sdk_version from linker flags

### PR DESCRIPTION
**Pull request by tjwio**
## Background ##
It seems like Xcode 11.4 linker doesn't allow `-sdk_version` and `-platform_version` or `-<platform>_version_min` flags to be passed together or it throws this warning:
```
ld: warning: -sdk_version and -platform_version are not compatible, ignoring -sdk_version
```
Couple of things worth noting:
1) same flags passed in Xcode 11.3 and below doesn't throw this warning
2) normal `xcodebuild` command doesn't pass `-sdk_version` at all and seemingly only relies on `-isysroot` and `-platform_version`
3) I couldn't find any references on llvm's or apple's llvm-project related to `-sdk_version`, but did see this change for Xcode 11, passing the new `platform_version` by default so it might make sense that they're deprecating `-sdk_version` usage in 11.4 https://reviews.llvm.org/D71579

## Changes ##
Remove `-sdk_version` from linker flags. It's seemingly not being used anymore and causing warnings with `-platform_version`.

## Notes ##
Mostly opening this PR to start the discussion and get more info from people who are more knowledgable in this area. I couldn't find anything related to changes to using `-sdk_version` flag in llvm from 11.3 to 11.4, but it's throwing warnings when using it now in 11.4 so something must've changed. Also, not an expert on the usage of the flag but I do know it's seemingly not being used in vanilla xcodebuilds, so I wanted to get more info on that flag and why it's being passed in the first place.

### CODE REVIEW

**Comment by v-jizhang** [comment]
@tjwio Please take a look at the failed test.
```
Failed target: //test/com/facebook/buck/apple/toolchain/impl:impl
FAIL com.facebook.buck.apple.toolchain.impl.AppleCxxPlatformsTest
```

File: src/com/facebook/buck/apple/toolchain/impl/AppleCxxPlatforms.java [hunk 1]
```diff
@@ -185,7 +185,6 @@ public static AppleCxxPlatform buildWithXcodeToolFinder(
     AppleConfig appleConfig = buckConfig.getView(AppleConfig.class);
 
     ImmutableList.Builder<String> ldflagsBuilder = ImmutableList.builder();
-    ldflagsBuilder.addAll(Linkers.iXlinker("-sdk_version", targetSdk.getVersion()));
```
[the remark this answers is not in the document]
Comment (tjwio) [reply src/com/facebook/buck/apple/toolchain/impl/AppleCxxPlatforms.java:188]: as far as I can tell, the new `-platform_version` flag already includes platform + sdk version, hence the warning
```
ld: warning: -sdk_version and -platform_version are not compatible, ignoring -sdk_version
```

[the remark this answers is not in the document]
Comment (v-jizhang) [reply src/com/facebook/buck/apple/toolchain/impl/AppleCxxPlatforms.java:188]: > as far as I can tell, the new `-platform_version` flag already includes platform + sdk version, hence the warning
> 
> ```
> ld: warning: -sdk_version and -platform_version are not compatible, ignoring -sdk_version
> ```
> 
> I think it's prob best to remove `-sdk_version` + `-mios-version-min` and replace it with the new `-platform_version <platform> <target_version> <sdk_version>`

I agree, do we have replacement already in buck code? If not, we should do it in this PR. Thank you.

[the remark this answers is not in the document]
Comment (tjwio) [reply src/com/facebook/buck/apple/toolchain/impl/AppleCxxPlatforms.java:188]: so upon further research, it looks like `clang` adds the `-platform_version` flag when it receives target and sysroot and when `-mlinker-version` >= 520.. so I think it's safe to remove `-sdk-version` for all newer versions of clang, but we may want to add a check to see if linker version is a certain min.
I tried looking around docs but couldn't find if there's an easy way to get clang/linker version, is that passed in somewhere accessible in that func `buildWithXcodeToolFinder`?

**Comment by v-jizhang** [comment]
Hi @tjwio, There is a commit to the dev branch that I think is what you are trying to do. Please take look.
Thanks.
 https://github.com/facebook/buck/commit/2dc13d9c83b78ca911fb31ec4de34c23fe59aeb4

**Comment by tjwio** [comment]
@v-jizhang looks solid from initial glance, will try it out with 11.4+ and report back 🙏

**Comment by tjwio** [comment]
@v-jizhang that fix lgtm thanks! feel free to close out this PR

**Comment by v-jizhang** [comment]
@tjwio Thanks

**Pull request closed without merging**